lib/src/data/datasource/local/DAOs/card_database.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import '../../../../core/util/constants_strings.dart';
+// ignore_for_file: avoid_dynamic_calls
+
+final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+final CollectionReference _firebaseCollection =
+    _firestore.collection(kFirestoreCollection);
+
+class CardDatabase {
+  Future<void> addItem({
+    required List<dynamic> cardList,
+    required String mainCollectionDoc,
+    required String subCollectionDoc,
+  }) async {
+    for (final card in cardList) {
+      final DocumentReference documentReferencer = _firebaseCollection
+          .doc(mainCollectionDoc)
+          .collection(subCollectionDoc)
+          .doc(
+            card![docRefCardId] as String,
+          );
+
+      documentReferencer.set(
+        card,
+        SetOptions(
+          merge: true,
+        ),
+      );
+    }
+  }
+
+  Future<QuerySnapshot> readItems({
+    required String mainCollectionDoc,
+    required String subCollectionDoc,
+  }) async {
+    _firestore.settings = const Settings(
+      persistenceEnabled: true,
+      cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
+    );
+    final CollectionReference cardItemCollection =
+        _firebaseCollection.doc(mainCollectionDoc).collection(subCollectionDoc);
```

Comment:
mainCollectionDoc is always the same in all calls?

Reply:
Changed it. Done.